src/os/bluestore/BlueStore.cc

```diff
@@ -1691,14 +1691,11 @@ void BlueStore::SharedBlob::get_ref(uint64_t offset, uint32_t length)
 
 void BlueStore::SharedBlob::put_ref(uint64_t offset, uint32_t length,
 				    PExtentVector *r,
-				    set<SharedBlob*> *maybe_unshared)
+				    SharedBlob** maybe_unshared)
 {
   assert(persistent);
   bool maybe = false;
-  persistent->ref_map.put(offset, length, r, maybe_unshared ? &maybe : nullptr);
-  if (maybe_unshared && maybe) {
-    maybe_unshared->insert(this);
-  }
+  persistent->ref_map.put(offset, length, r, maybe_unshared && !*maybe_unshared? &maybe : nullptr);
```

Comment:
It could be that taking `SharedBlob** maybe_unshared` is unnecessary. All `SharedBlob::put_ref` is doing now is just calling `bluestore_extent_ref_map_t::put` -- the `maybe` value isn't used internally nor exposed outside. Is that intended?

Reply:
Missed maybe_unshared assignment here. Fixing. Thanks.